makefile

```
@@ -44,52 +75,43 @@ validate-psql-connection:
 	pg_isready -h $(DOCKER_DNS) -p 5432
 
 
-env-setup:
-	npm ci
-	-docker-compose up -d postgres localstack
-	cp .env.example .env
-	while ! make validate-psql-connection validate-aws-connection; \
-		do echo "waiting for postgres, localstack bootup"; \
-		sleep 0.5; \
-		done
+test-tenant: TENANT = 'test'
+test-tenant: tenant
 
-test-tenant:
-	make tenant TENANT='test'
+dev-tenant: TENANT = 'dev'
+dev-tenant: tenant
 
-dev-tenant:
-	make tenant TENANT='dev'
-
-ci-setup: env-setup test-tenant
-	npm ci --loglevel=error
-	make run -e DOCKER_DNS=$(DOCKER_DNS) 2>&1 | tee test_logs &
-	while ! grep -q "starting in Actix" test_logs; \
-		do echo "ci-test: waiting for bootup..." && sleep 4; \
-		done
-	# NOTE: `make db-init` finally starts a postgres container and runs all the migrations with locked-schema option
-	# to prevent update of schema.rs for both cac and experimentation.
-	# NOTE: The container spinned-up here is the actual container being used in development
-	echo setup completed successfully!!!
-
-setup: env-setup
+SETUP_DEPS = env-file db localstack
+ifdef CI
+	SETUP_DEPS += test-tenant
+endif
+setup: $(SETUP_DEPS)
+	npm ci
 
 kill:
-	-pkill -f target/debug/superposition &
+	-@pkill -f target/debug/superposition &
 
 get-password:
 	export DB_PASSWORD=`./docker-compose/localstack/get_db_password.sh` && echo $$DB_PASSWORD
 
+superposition: CARGO_FLAGS += --features=$(FEATURES)
 superposition:
-	cargo run --color always --bin superposition --no-default-features --features=$(FEATURES)
+	cargo build $(CARGO_FLAGS) --bin superposition
 
 superposition-example:
 	cargo run --bin cac-demo-app
 
+superposition_legacy: CARGO_FLAGS += --features='ssr superposition_types/disable_db_data_validation
+superposition_legacy: CARGO_FLAGS += superposition_types/disable_db_data_validation
+superposition_legacy: CARGO_FLAGS += context_aware_config/disable_db_data_validation
+superposition_legacy: CARGO_FLAGS += experimentation_platform/disable_db_data_validation'
 superposition_legacy:
-	cargo run --color always --bin superposition --no-default-features --features='ssr superposition_types/disable_db_data_validation context_aware_config/disable_db_data_validation experimentation_platform/disable_db_data_validation'
+	cargo run $(CARGO_FLAGS) --bin superposition
```

lets keep it same as `superposition`
```suggestion
	cargo build $(CARGO_FLAGS) --bin superposition
```